src/views/UserForm/UserForm.tsx

```diff
@@ -1,65 +1,33 @@
-import React, { useState } from 'react'
+import React from 'react'
 
-import { QuestionConditionTypes } from 'utils/helpers'
 import { Question } from 'common/models'
 
-import FormBlock from './components/UserQuestion'
+import StateWrapper from './components/StateWrapper'
+import { checkForCondition } from './checkForCondition'
 
 interface UserFormProps {
     questions: Question[]
     parentValue?: string
 }
 
-function UserForm({ questions, parentValue }: UserFormProps) {
-    const [formInputValue, setFormInputValue] = useState('')
-
-    const checkCondition = (
-        conditionType: string,
-        conditionValue: string,
-        level: number,
-        formInputValue?: string
-    ) => {
-        if (level === 0) return true
-
-        //eslint-disable-next-line
-        if (conditionType === QuestionConditionTypes.Equals && formInputValue == conditionValue)
-            return true
-        if (
-            conditionType === QuestionConditionTypes.Less &&
-            formInputValue !== '' &&
-            parseInt(formInputValue!) < parseInt(conditionValue, 10)
-        )
-            return true
-        if (
-            conditionType === QuestionConditionTypes.Greater &&
-            parseInt(formInputValue!) > parseInt(conditionValue, 10)
-        )
-            return true
-
-        return false
-    }
-
+const UserForm: React.SFC<UserFormProps> = ({ questions, parentValue }): JSX.Element => {
     return (
         <>
-            {questions.map(e => (
-                <div key={e.id}>
-                    {checkCondition(e.conditionType!, e.conditionValue!, e.level, parentValue) && (
-                        <>
-                            <FormBlock
-                                data={e}
-                                setFormInputValue={({
-                                    target,
-                                }: React.ChangeEvent<HTMLInputElement>) =>
-                                    setFormInputValue(target.value)
-                                }
-                            />
-                            {e.children && (
-                                <UserForm questions={e.children} parentValue={formInputValue} />
-                            )}
-                        </>
-                    )}
-                </div>
-            ))}
+            {questions.map(
+                question =>
+                    checkForCondition(
+                        question.conditionType!,
```

Comment:
you've just typed question to have those properties optional, maybe try passiing object as one param